libmscore/mmrest.cpp

```diff
@@ -0,0 +1,275 @@
+    m_symsWidth = r.m_symsWidth;
+}
+
+//---------------------------------------------------------
+//   MMRest::draw
+//---------------------------------------------------------
+
+void MMRest::draw(QPainter* painter) const
+{
+    if (shouldNotBeDrawn() || (track() % VOICES)) {     //only on voice 1
+        return;
+    }
+
+    qreal _spatium = spatium();
+
+    // draw number
+    painter->setPen(curColor());
+    std::vector<SymId>&& numberSym = toTimeSigString(QString("%1").arg(m_number));
+    QRectF numberBox = symBbox(numberSym);
+    qreal x = (m_width - numberBox.width()) * .5;
+    qreal y = m_numberPos * spatium() - staff()->height() * .5;
+    drawSymbols(numberSym, painter, QPointF(x, y));
+
+    if (score()->styleB(Sid::oldStyleMultiMeasureRests)
+      && m_number <= score()->styleI(Sid::mmRestOldStyleMaxMeasures)) {
+        x = (m_width - m_symsWidth) * 0.5;
+        drawSymbols(m_restSyms, painter, QPointF(x, 0), 1.0, SPACING);
+        if (m_isOddNumber) {
+            // add the whole rest separately, because otherwise it would draw too low
+            SymId sym = SymId::restWhole;
+            x += m_symsWidth;
+            x -= symBbox(sym).width();
+            drawSymbol(sym, painter, QPointF(x, -_spatium));
+        }
+    } else {
+        // draw horizontal line
+        qreal hBarThickness = score()->styleP(Sid::mmRestHBarThickness);
+        qreal halfHBarThickness = hBarThickness * .5;
+        QPen pen(painter->pen());
+        pen.setCapStyle(Qt::FlatCap);
+        pen.setWidthF(hBarThickness);
+        painter->setPen(pen);
+        if (score()->styleB(Sid::mmRestNumberMaskHBar)     // avoid painting line through number
+          && (y + (numberBox.height() * .5)) > -halfHBarThickness
+          && (y - (numberBox.height() * .5)) < halfHBarThickness) {
+            qreal gapDistance = (numberBox.width() + _spatium) * .5;
+            qreal midpoint = m_width * .5;
+            painter->drawLine(QLineF(0.0, 0.0, midpoint - gapDistance, 0.0));
+            painter->drawLine(QLineF(midpoint + gapDistance, 0.0, m_width, 0.0));
+        } else {
+            painter->drawLine(QLineF(0.0, 0.0, m_width, 0.0));
+        }
+
+        // draw vertical lines
+        qreal vStrokeThickness = score()->styleP(Sid::mmRestHBarVStrokeThickness);
+        if (vStrokeThickness) { // don't draw at all if 0, QPainter interprets 0 pen width differently
+            pen.setWidthF(vStrokeThickness);
+            painter->setPen(pen);
+            qreal halfVStrokeHeight = score()->styleP(Sid::mmRestHBarVStrokeHeight) * .5;
+            painter->drawLine(QLineF(0.0, -halfVStrokeHeight, 0.0, halfVStrokeHeight));
+            painter->drawLine(QLineF(m_width, -halfVStrokeHeight, m_width,  halfVStrokeHeight));
+        }
+    }
+}
+
+//---------------------------------------------------------
+//   layout
+//---------------------------------------------------------
+
+void MMRest::layout()
+{
+    m_number = measure()->mmRestCount();
```

Comment:
shouldn't this `m_number` be set in the constructor? It seems illogical to me for it to be handled by `layout()`.
Getters and setters?

Reply:
No, this number needs to be able to change on each relayout, and the object is not destroyed and newly created when the length gets larger or smaller.